src/CoreBluetooth/AdvertisementData.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Linq;
+
+using XamCore.ObjCRuntime;
+using XamCore.Foundation;
+using XamCore.CoreFoundation;
+
+namespace XamCore.CoreBluetooth {
+
+	public partial class AdvertisementData {
+		
+		public IDictionary <CBUUID, NSData> ServiceData {
+			get {
+				using (var nsDict = _ServiceData) {
```

Comment:
Defining the StrongDict as follows:

	[StrongDictionary ("AdvertisementDataKeys")]
	interface AdvertisementData {
		string LocalName { get; set; }
		NSData ManufacturerData { get; set; }
		NSDictionary <CBUUID, NSData > ServiceData { get; set; }
		CBUUID [] ServiceUuids { get; set; }
		CBUUID [] OverflowServiceUuids { get; set; }
		NSNumber TxPowerLevel { get; set; }
		bool IsConnectable { get; set; }
		CBUUID [] SolicitedServiceUuids { get; set; }
	}

Throws the following compiler error:

/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/compat/CoreBluetooth/AdvertisementData.g.cs(59,96): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/compat/CoreBluetooth/AdvertisementData.g.cs(59,96): error CS1525: Unexpected symbol `2'
Compilation failed: 2 error(s), 0 warnings
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/tvos/tvos/CoreBluetooth/AdvertisementData.g.cs(60,88): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/tvos/tvos/CoreBluetooth/AdvertisementData.g.cs(60,88): error CS1525: Unexpected symbol `2'
Compilation failed: 2 error(s), 0 warnings
make[1]: *** [build/mac/compat/XamMac.dll] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [build/tvos/tvos-64/Xamarin.TVOS.dll] Error 1
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/mobile/CoreBluetooth/AdvertisementData.g.cs(61,88): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/mobile/CoreBluetooth/AdvertisementData.g.cs(61,88): error CS1525: Unexpected symbol `2'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/mobile/CoreBluetooth/AdvertisementData.g.cs(61,88): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/mobile/CoreBluetooth/AdvertisementData.g.cs(61,88): error CS1525: Unexpected symbol `2'
Compilation failed: 2 error(s), 0 warnings
Compilation failed: 2 error(s), 0 warnings
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/full/CoreBluetooth/AdvertisementData.g.cs(60,88): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/full/CoreBluetooth/AdvertisementData.g.cs(60,88): error CS1525: Unexpected symbol `2'
Compilation failed: 2 error(s), 0 warnings
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/full/CoreBluetooth/AdvertisementData.g.cs(60,88): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/mac/full/CoreBluetooth/AdvertisementData.g.cs(60,88): error CS1525: Unexpected symbol `2'
Compilation failed: 2 error(s), 0 warnings
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/ios/compat/CoreBluetooth/AdvertisementData.g.cs(65,98): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/ios/compat/CoreBluetooth/AdvertisementData.g.cs(65,98): error CS1525: Unexpected symbol `2'
Compilation failed: 2 error(s), 0 warnings
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/ios/native/CoreBluetooth/AdvertisementData.g.cs(65,88): error CS1056: Unexpected character ``'
/Users/mandel/Xamarin/maccore-opensource/xamarin-macios/src/build/ios/native/CoreBluetooth/AdvertisementData.g.cs(65,88): error CS1525: Unexpected symbol `2'

The generated code is as follows:

		public NSDictionary<CBUUID, NSData> ServiceData {
			get {
				return Dictionary [AdvertisementDataKeys.ServiceDataKey] as Foundation.NSDictionary`2[CoreBluetooth.CBUUID,Foundation.NSData];
			}
			set {
				SetNativeValue (AdvertisementDataKeys.ServiceDataKey, value);
			}
		}

@dalexsoto, @spouliot I can always fix the generator to deal with the fact of using templated classes. Which is probably the right fix.